mergify_engine/web/api/security.py

```diff
@@ -85,3 +88,45 @@ async def get_installation(
         )
     except exceptions.MergifyNotInstalled:
         raise fastapi.HTTPException(status_code=403)
+
+
+async def get_queue_freeze_feature(
+    owner: github_types.GitHubLogin = fastapi.Path(  # noqa: B008
+        ..., description="The owner of the repository"
+    ),
+    repository: github_types.GitHubRepositoryName = fastapi.Path(  # noqa: B008
+        ..., description="The name of the repository"
+    ),
+    redis_cache: utils.RedisCache = fastapi.Depends(  # noqa: B008
+        redis.get_redis_cache
+    ),
+    redis_queue: utils.RedisQueue = fastapi.Depends(  # noqa: B008
+        redis.get_redis_queue
+    ),
+    installation_json: github_types.GitHubInstallation = fastapi.Depends(  # noqa: B008
+        get_installation
+    ),
+) -> subscription.Features:
+
+    async with github.aget_client(installation_json) as client:
+        try:
+            # Check this token has access to this repository
+            repo = typing.cast(
+                github_types.GitHubRepository,
+                await client.item(f"/repos/{owner}/{repository}"),
+            )
+        except (http.HTTPNotFound, http.HTTPForbidden, http.HTTPUnauthorized):
+            raise fastapi.HTTPException(status_code=404)
+
+        sub = await subscription.Subscription.get_subscription(
+            redis_cache, repo["owner"]["id"]
+        )
+
+        installation = context.Installation(
+            installation_json, sub, client, redis_cache, redis_queue
+        )
+
+        if installation.subscription.has_feature(subscription.Features.QUEUE_FREEZE):
+            return subscription.Features.QUEUE_FREEZE
```

Comment:
Actually, you should return the `installation` here or the like.

The problem is that your current code gets twice the subscription for example, here, and in `get_respository_context`; this is quite suboptimal.

So I think if this function returns the `installation` that it built, you can then pass it to `get_repository_context` and save some code — or you don't even need `get_repository_context` maybe?